OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -124,7 +125,11 @@ public Aircraft(ActorInitializer init, AircraftInfo info)
 			IsPlane = !info.CanHover;
 		}
 
-		public void Created(Actor self) { um = self.TraitOrDefault<UpgradeManager>(); }
+		public void Created(Actor self)
+		{
+			um = self.TraitOrDefault<UpgradeManager>();
+			speedModifiers = self.TraitsImplementing<ISpeedModifier>().ToArray().Select(sm => sm.GetSpeedModifier());
```

Comment:
The enumerable is passed directly into `ApplyPercentageModifiers` which does the heavy lifting.

So all we're doing is passing a field directly to a method - it doesn't come cheaper than that.